### PR TITLE
Implement --strict-module-detection

### LINKED ISSUE #77: Allow Explicitly Running/Not Running Code as a Module
Like ```--module```/```-m``` and ```--no-module```

### PR DESCRIPTION
**Pull request by TheBrokenRail**
Fixes #77

### CODE REVIEW

File: src/cli.c [hunk 1]
```diff
@@ -134,7 +139,11 @@ int main(int argc, char **argv) {
         } else {
             const char *filename;
             filename = argv[optind];
-            JSValue ret = TJS_EvalFile(ctx, filename, JS_EVAL_TYPE_MODULE, true);
+            int flags = JS_EVAL_TYPE_MODULE;
+            if (strict_module_detection && has_suffix(filename, ".mjs")) {
```
Comment (saghul) [inline src/cli.c]: Isn't this the opposite of what you want here? you probably want `!has_suffix`

Comment (saghul) [inline src/cli.c]:
```suggestion
            if (strict_module_detection && !has_suffix(filename, ".mjs")) {
```

**Comment by saghul** [comment]
This needs a rebase too.

File: src/cli.c (hunk 1 above)
Comment (TheBrokenRail) [reply src/cli.c]: Fixed!

**Comment by TheBrokenRail** [comment]
Rebased!

**Comment by TheBrokenRail** [comment]
When will this be merged?

**Pull request merged**